### PR TITLE
LPS-203001 Integration tests failing on Oracle Database

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-203001](https://liferay.atlassian.net/browse/LPS-203001)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceSearchObjectEntriesTest.java [hunk 1]
```diff
@@ -150,11 +150,11 @@ public void testBigDecimalKeyword() throws Exception {
 		_assertKeywords("[44.9999 TO 45.1111]", 0);
 		_assertKeywords("4", 1);
 		_assertKeywords("45", 1);
-		_assertKeywords("45.0000", 1);
+		_assertKeywords("45.0000", 0);
```
Comment (p-albuquerque) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceSearchObjectEntriesTest.java:153]: If do you change to:
`_assertKeywords("45", 1);`
It works too?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceSearchObjectEntriesTest.java [hunk 2]
```diff
@@ -150,11 +150,11 @@ public void testBigDecimalKeyword() throws Exception {
...
 		_assertKeywords("45.0001", 0);
 		_assertKeywords("bravo 4 charlie", 1);
 		_assertKeywords("bravo 45 charlie", 1);
-		_assertKeywords("bravo 45.0 charlie", 1);
+		_assertKeywords("bravo 45.0 charlie", 0);
```
Comment (p-albuquerque) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceSearchObjectEntriesTest.java:157]: If do you change to:
`_assertKeywords("bravo 45 charlie", 1);`
It works too?

**Review by p-albuquerque — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 3]
```diff
@@ -2832,6 +2833,17 @@ private void _assertCount(int count) throws Exception {
...
 
+	private void _assertDate(Date expectedDate, Date actualDate) {
+		LocalDate expectedLocalDate = LocalDateTimeUtil.toLocalDateTime(
+			expectedDate
+		).toLocalDate();
+		LocalDate actualLocalDate = LocalDateTimeUtil.toLocalDateTime(
+			actualDate
+		).toLocalDate();
+
+		Assert.assertTrue(expectedLocalDate.isEqual(actualLocalDate));
```
[the remark this answers is not in the document]
Comment (pedro-oliveira446) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java]: Just did.

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:object-db-integration

**Review by p-albuquerque — APPROVED** [review]

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java [hunk 4]
```diff
@@ -1757,9 +1764,31 @@ private Map<String, Serializable> _toObjectValues(
...
 
-				values.put(
-					objectField.getName(),
-					_toDate(locale, String.valueOf(value)));
+				String dateValue = String.valueOf(value);
+
+				if (!StringUtil.equals(objectField.getName(), "createDate") &&
+					!StringUtil.equals(objectField.getName(), "modifiedDate")) {
+
+					String pattern = ObjectFieldUtil.getDateTimePattern(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:1772]: It seems like a lot of code to do a common thing, I imagine other teams have had this need before, so maybe these classes/methods below can help you write less code. Can you please try this?
`DateUtil, DateFormatFactoryUtil, GetterUtil#getDate(java.lang.String, java.text.DateFormat)`
This [link](https://www.baeldung.com/java-date-without-time) can help too.

Another point is that maybe the right solution is to override `ObjectFieldBusinessType#getValue` method in `DateObjectFieldBusinessType` class and move this logic to there, what do you think?
Maybe we can submit this specific fix in another pull request so as not to block others and also verify the impact of this change by running tests only with it.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]
Hi @pedro-oliveira446, good work here, I left some small comments, please let me know if you need help with this.

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:object-db-integration

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:object-db-integration

**Review by guilhermedcamacho — APPROVED** [review]
LGTM :tada:

**Comment by guilhermedcamacho** [comment]
Hey @pedro-oliveira446, we should check the CI results carefully, as some of your changes may cause side effects.

**Comment by pedro-oliveira446** [comment]
Hey @guilhermedcamacho,
I've analyzed it and everything looks right on CI, I just added a 'Semver' commit.

**Comment by pedro-oliveira446** [comment]
Resent [here](https://github.com/brianchandotcom/liferay-portal/pull/145511)

**Pull request closed without merging**